src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTests.scala

```diff
@@ -409,76 +410,74 @@ class E2EHyperspaceRulesTests extends HyperspaceSuite with SQLHelper {
       sortedRowsWithHyperspaceDisabled.sameElements(getSortedRows(dfAfterHyperspaceDisabled)))
   }
 
-  test("Verify Join Index Rule utilizes indexes correctly after incremental refresh.") {
+  test("Verify JoinIndexRule utilizes indexes correctly after incremental refresh.") {
     withTempPathAsString { testPath =>
-      withSQLConf(IndexConstants.REFRESH_APPEND_ENABLED -> "true") {
-        // Setup. Create data.
-        val indexConfig = IndexConfig("index", Seq("c2"), Seq("c4"))
-        import spark.implicits._
-        SampleData.testData
-          .toDF("c1", "c2", "c3", "c4", "c5")
-          .limit(10)
-          .write
-          .parquet(testPath)
-        val df = spark.read.load(testPath)
-
-        // Create index.
-        hyperspace.createIndex(df, indexConfig)
-
-        // Append to original data.
-        SampleData.testData
-          .toDF("c1", "c2", "c3", "c4", "c5")
-          .limit(3)
-          .write
-          .mode("append")
-          .parquet(testPath)
-
-        // Refresh index.
-        hyperspace.refreshIndex(indexConfig.indexName)
-
-        // Create a join query.
-        val leftDf = spark.read.parquet(testPath)
-        val rightDf = spark.read.parquet(testPath)
-
-        def query(): DataFrame = {
-          leftDf
-            .join(rightDf, leftDf("c2") === rightDf("c2"))
-            .select(leftDf("c2"), rightDf("c4"))
-        }
+      // Setup. Create data.
+      val indexConfig = IndexConfig("index", Seq("c2"), Seq("c4"))
+      import spark.implicits._
+      SampleData.testData
+        .toDF("c1", "c2", "c3", "c4", "c5")
+        .limit(10)
+        .write
+        .parquet(testPath)
+      val df = spark.read.load(testPath)
+
+      // Create index.
+      hyperspace.createIndex(df, indexConfig)
 
-        // Verify indexes are used, and all index files are picked.
-        verifyIndexUsage(
-          query,
-          getIndexFilesPath(indexConfig.indexName, Seq(0, 1)) ++
-            getIndexFilesPath(indexConfig.indexName, Seq(0, 1)))
-
-        // With Hyperspace disabled, verify there are shuffle and sort nodes as expected.
-        spark.disableHyperspace()
-        val dfWithHyperspaceDisabled = query()
-        var shuffleNodes = dfWithHyperspaceDisabled.queryExecution.executedPlan.collect {
-          case s: ShuffleExchangeExec => s
-        }
-        assert(shuffleNodes.size == 2)
-        var sortNodes = dfWithHyperspaceDisabled.queryExecution.executedPlan.collect {
-          case s: SortExec => s
-        }
-        assert(sortNodes.size == 2)
-
-        // With Hyperspace enabled, verify bucketing works as expected. This is reflected in
-        // shuffle nodes being eliminated.
-        spark.enableHyperspace()
-        val dfWithHyperspaceEnabled = query()
-        shuffleNodes = dfWithHyperspaceEnabled.queryExecution.executedPlan.collect {
-          case s: ShuffleExchangeExec => s
-        }
-        assert(shuffleNodes.isEmpty)
+      // Append to original data.
```

Comment:
Could you also test with deleted files?
and also could you add some e2e tests using refreshed index? and compare the actual query results w/ or w/o index?

Reply:
thanks @sezruby , please check this existing test https://github.com/microsoft/hyperspace/blob/master/src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTests.scala#L493

Reply:
added test to verify data